pandas/tests/frame/indexing/test_where.py

```diff
@@ -787,3 +791,30 @@ def test_where_columns_casting():
     result = df.where(pd.notnull(df), None)
     # make sure dtypes don't change
     tm.assert_frame_equal(expected, result)
+
+
+optional_ints = st.lists(st.one_of(st.integers(), st.none()), max_size=10, min_size=3)
```

Comment:
ideally would want to add lots of other things here, e.g. datetetimes (naive & tz-aware), timedelta, categorical, period, interval. (can be a followup)

also would be better to centralize some of this hypothesis type generation. (do this in this PR)